File: python/problem-string-palindrome/palindrome_number.py
#   https://leetcode.com/problems/palindrome-number
#   89.85%


class Solution:
    def isPalindrome(self, x):
        s = str(x)
        return s == s[::-1]


s = Solution()
data = [(121, True), (-121, False)]
for num, expected in data:
    real = s.isPalindrome(num)
    print('{} expected {} real {} result {}'.format(num, expected, real, expected == real))

File: elixir/problem-string-palindrome/palindrome_number.exs
# https://leetcode.com/problems/palindrome-number/

defmodule Solution do
  # runtime; 26.19%
  # memory; 88.9%
  @spec is_palindrome(x :: integer) :: boolean
  def is_palindrome(x) do
    Integer.to_string(x) == Integer.to_string(x) |> String.reverse
  end
end

data = [{121, true},
        {-121, false},
        {10, false},
        ]
for {x, expect} <- data do
  real = Solution.is_palindrome(x)
  IO.puts "#{x} expect #{expect} real #{real} result #{expect == real}"
end